scenario/consistency_checker.py

```diff
@@ -565,3 +566,22 @@ def check_containers_consistency(
         errors.append(f"Duplicate container name(s): {dupes}.")
 
     return Results(errors, [])
+
+
+def check_cloudspec_consistency(
+    *,
+    state: "State",
+    event: "Event",
+    charm_spec: "_CharmSpec",
+    **_kwargs,  # noqa: U101
+) -> Results:
+    """Check that Kubernetes charms/models don't have `state.cloud_spec`."""
+
+    errors, warnings = [], []
```

Comment:
There's not really any need to have this on one line, and not doing so avoids the extra work (theoretically building and unpacking a tuple, although in practice I think the interpreter will use the `swap` opcode to do it).

```suggestion
    errors = []
    warnings = []
```

Reply:
I'm pushing this back, reasons:

1. "There's not really any need to have this on one line" isn't a solid argument. This logic cuts both ways: There is no need to have it on two lines.

2. More pythonic and elegant. I wouldn't do this:

`a, b, c, d, e, f, g = True, True, True, True, True, False, True`

But declaring two short variables with zero values in the same line is more than reasonable since it doesn't hurt readability (might even improve it).

3. Technically it hurts performance, but in the real word it doesn't make a difference:

```python3
import timeit

code1 = "a, b = [], []"
code2 = """a = []
b = []"""

print(timeit.timeit(stmt=code1, number=1000))
# 3.6041000000000545e-05

print(timeit.timeit(stmt=code2, number=1000))
# 3.066699999999985e-05
```

Reply:
> 1. "There's not really any need to have this on one line" isn't a solid argument. This logic cuts both ways: There is no need to have it on two lines.

It's objectively more readable on two lines.

> 2. More pythonic and elegant.

Two lines is definitely more Pythonic. Elegant is very subjective.

> 3. Technically it hurts performance, but in the real word it doesn't make a difference:

I don't agree that just because the performance impact isn't noticeable that it's not worth having a preference for the more efficient code.

I suggest @PietroPasotti makes this call :smile:.

Reply:
Let's be a bit careful about making bold statements like "it's objectively more readable on two lines" without an argument or evidence to back that up. I also agree with Tiexin that "there's not really any need to have this on one line" isn't by itself a solid argument.

I think there *are* arguments one could make (eg: it's easier to see the variable name when it's hard on the left, rather than hidden in the middle of a line). If we wanted to spend the time, it might also be worth looking at a bunch of Python projects to see if they use this form.

I don't think it's worth it just for this, though. More important is *precedent* in this project, specifically in this file:

```
$ rg 'errors = \[\]' -A1
scenario/consistency_checker.py
61:    errors = []
62-    warnings = []
--
104:    errors = []
105-    warnings = []
--
129:    errors = []
130-    warnings = []
...
```

So I think that answers our question. Pietro's precedent (in this same file) is to do this on two lines, so we should too.

Reply:
yup yup, project consistency takes precedence :)

Reply:
Thanks for the comments. Subjective/objective aside, I will change it to two lines for consistency because if there is anything I love more than cycling it's consistency. If I had an English name, Consistency would be my middle name :)